#! /bin/sh /usr/share/dpatch/dpatch-run
## routing_sec_patch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixed a security fix by upstream.

@DPATCH@
--- a/actionpack/lib/action_controller/routing.rb	2006-08-10 13:10:33.000000000 -0500
+++ b/actionpack/lib/action_controller/routing.rb	2006-08-10 13:13:26.000000000 -0500
@@ -273,7 +273,7 @@
             $LOAD_PATH.select do |base|
               base = File.expand_path(base)
               extended_root = File.expand_path(RAILS_ROOT)
-              base.match(/\A#{Regexp.escape(extended_root)}\/*#{file_kinds(:lib) * '|'}/) || base =~ %r{rails-[\d.]+/builtin}
+              base.match(/\A#{Regexp.escape(extended_root)}\/*(#{file_kinds(:lib) * '|'})/) || base =~ %r{rails-[\d.]+/builtin}
             end
           else
             $LOAD_PATH
